sdk/src/main/java/io/dapr/actors/runtime/ActorRuntime.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+package io.dapr.actors.runtime;
+
+import java.util.HashMap;
+import java.util.Set;
+import java.util.function.Function;
+
+import io.dapr.actors.*;
+
+/**
+ * Contains methods to register actor types. Registering the types allows the runtime to create instances of the actor.
+ */
+public class ActorRuntime {
+    private static volatile ActorRuntime instance;
+    private static AppToDaprAsyncClient appToDaprHttpAsyncClient;
+    private final String TraceType = "ActorRuntime";
+    private final HashMap<String, ActorManager> actorManagers;
+
+    private ActorRuntime() throws IllegalStateException{
+        if (instance != null) {
+            throw new IllegalStateException("ActorRuntime should only be constructed once");
+        }
+
+        this.actorManagers = new HashMap<String, ActorManager>();
+        appToDaprHttpAsyncClient = new DaprClientBuilder().buildAsyncClient();
+    }
+
+    /**
+     * Returns an ActorRuntime object.
+     * @return An ActorRuntime object.
+     */
+    public static ActorRuntime getInstance() {
+        if (instance == null) {
+            synchronized (ActorRuntime.class) {
+                if (instance == null) {
+                    instance = new ActorRuntime();
+                }
+            }
+        }
+
+        return  instance;
+    }
+
+    /**
+     *
+     * @return Actor type names registered with the runtime.
+     */
+    public Set<String> getRegisteredActorTypes() {
+        return this.actorManagers.keySet();
+    }
+
+    /**
+     * Registers an actor with the runtime.
+     * @param clazz The type of actor.
+     */
+    <TActor extends AbstractActor> void RegisterActor(Class<TActor> clazz) {
```

Comment:
I usually just see people using T, unless you have more than one, single letter is enough.